ElrondNetwork/elrond-go
# EN-12707: improve sc generation for system vm calls

## Description of the reasoning behind the pull request (what feature was missing / how the problem was manifesting itself / what was the motive behind the refactoring)
- the SCRs resulted after system vm calls were misleading (for example, when creating a new delegation contract) 
  
## Proposed Changes
- set `SenderAddress` for `outputTransfer` inside `eei.go` if the flag is enabled. This will help generate SCRs in the correct way 

## Testing procedure
Start a testnet and create a new delegation contract ([docs](https://docs.elrond.com/validators/delegation-manager/#creating-a-new-delegation-contract)). after this, you should see 3 SCRs:
- `erd1qqqqqqqqqqqqqqqpqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqylllslmq6y6` -> `CONTRACT_ADDRESS` 1250 egld
- `CONTRACT_ADDRESS` -> `erd1qqqqqqqqqqqqqqqpqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqplllst77y4l` 1250 egld
- scr for gas refund (optional)
In comparison, here's an example on how the SCRs were generated before: [explorer](https://explorer.elrond.com/transactions/7f21cda7076dd401302e4336cedef5273be77ba2aa98153da83ed978ced75d52)
